internal/buf/pool.go

```diff
@@ -107,26 +109,71 @@ func (s *segment) allocate(count int) (Buf, bool) {
 // If no segment has available capacity, the pool waits a few times until memory becomes released
 // and falls back to allocating from the heap.
 type Pool struct {
-	maxSegments     int
-	segmentReleased chan struct{} // channel which is notified whenever any segment becomes available for allocations
+	totalAllocatedBytes   int64
+	totalReleasedBytes    int64
+	totalAllocatedBuffers int64
+	totalReleasedBuffers  int64
+
+	poolID string
+
+	closed chan struct{}
+
+	tagMutators []tag.Mutator
 
 	// this protects the slice, to be able to atomically replace it
 	mu          sync.Mutex
 	segmentSize int
 	segments    []*segment
 }
 
-// NewPool creates a buffer pool, composed of N fixed-length segments of specified maximum size.
-func NewPool(segmentSize, maxSegments int) *Pool {
+// NewPool creates a buffer pool, composed of fixed-length segments of specified maximum size.
+func NewPool(segmentSize int, poolID string) *Pool {
 	p := &Pool{
-		segmentSize:     segmentSize,
-		segmentReleased: make(chan struct{}),
-		maxSegments:     maxSegments,
+		poolID:      poolID,
+		tagMutators: []tag.Mutator{tag.Insert(tagKeyPool, poolID)},
+		segmentSize: segmentSize,
+		closed:      make(chan struct{}),
 	}
 
+	go func() {
+		for {
+			select {
+			case <-p.closed:
+				return
+
+			case <-time.After(1 * time.Second):
+				p.reportMetrics()
+			}
+		}
+	}()
+
 	return p
 }
 
+// Close closes the pool
+func (p *Pool) Close() {
+	close(p.closed)
+}
+
+func (p *Pool) reportMetrics() {
```

Comment:
nit: pass the `Context` from the caller (go routine above)?

Reply:
done